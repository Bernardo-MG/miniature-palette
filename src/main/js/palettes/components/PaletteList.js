import React, { Fragment, useState } from 'react';

import PropTypes from 'prop-types';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardHeader from '@material-ui/core/CardHeader';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';

import ArrowBackIcon from '@material-ui/icons/ArrowBack';

function PaletteData({ data, onReturn }) {
   return <Fragment>
      <Grid item xs={8}>
         <Card>
            <CardHeader
               title={ <Typography>{data.name}</Typography> }
               action={
                  <IconButton edge="end" aria-label="back" onClick={onReturn}>
                     <ArrowBackIcon />
                  </IconButton>
               } />
            <CardContent>
               <List>
                  {data.paints.map((paint) =>
                     <ListItem key={paint.name}>
                        <ListItemText primary={paint.name} secondary={paint.code} />
                     </ListItem>
                  )}
               </List>
            </CardContent>
         </Card>
      </Grid>
   </Fragment>;
}

PaletteData.propTypes = {
   data: PropTypes.shape({
      name: PropTypes.string,
      paints: PropTypes.array
   }).isRequired,
   onReturn: PropTypes.func.isRequired
};

function PaletteList({ data }) {
   const [selected, setSelected] = useState(null);
   let paletteData;

   if (selected) {
      // Selected group
      paletteData = <Fragment>
         <PaletteData data={selected} onReturn={() => setSelected(null)} />
      </Fragment>;
   } else {
      // List of groups
      paletteData = <List>
         {data.map((palette) =>
            <ListItem button key={palette.name} onClick={() => setSelected(palette)}>
               <ListItemText primary={palette.name}/>
            </ListItem>
         )}
      </List>;
   }

   return <Fragment>
      <Grid container spacing={3}>
         { paletteData }
      </Grid>
   </Fragment>;
}

PaletteList.propTypes = {
   data: PropTypes.array.isRequired
};

export default PaletteList;
