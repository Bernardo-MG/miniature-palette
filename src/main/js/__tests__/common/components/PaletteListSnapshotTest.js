import React from 'react'
import renderer from 'react-test-renderer';

import PaletteList from 'palettes/components/PaletteList';

describe('<PaletteList />', () => {
   it('renders correctly when there is no data', () => {
      const tree = renderer
         .create(<PaletteList
               data={[{ name: '', paints: [] }]}
               onEdit={ () => 'action' } />)
         .toJSON();
      expect(tree).toMatchSnapshot();
   }),
   it('renders correctly with a full palette', () => {
      const tree = renderer
         .create(<PaletteList
               data={[{ id: 1, name: 'name', paints: [ {name: 'paint'} ] }]}
               onEdit={ () => 'action' } />)
         .toJSON();
      expect(tree).toMatchSnapshot();
   })
});
