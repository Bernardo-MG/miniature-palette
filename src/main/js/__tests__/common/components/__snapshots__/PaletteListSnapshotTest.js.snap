// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaletteList /> renders correctly when there is no data 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          
        </span>
      </div>
    </div>
  </ul>
</div>
`;

exports[`<PaletteList /> renders correctly with a full palette 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
        >
          name
        </span>
      </div>
      <span
        className="MuiTouchRipple-root"
      />
    </div>
  </ul>
</div>
`;
